test/extended/networking/util.go

```diff
@@ -862,6 +864,20 @@ func areMachineConfigPoolsReadyWithIPsec(oc *exutil.CLI) (bool, error) {
 	return masterWithIPsec && workerWithIPsec, nil
 }
 
+func areMachineConfigPoolsReadyWithoutIPsec(oc *exutil.CLI) (bool, error) {
```

Comment:
There is something in this code that makes it very difficult to read. Some suggestions:

* Remove method `areWorkerMachineConfigPoolsReady`, just as you don't have a `areMasterMachineConfigPoolsReady`
* s/`areMachineConfigPoolReadyWithoutMachineConfig`/`areMachineConfigPoolsReadyWithoutMachineConfig`
* s/`isMachineConfigReadyInPools`/`areMachineConfigPoolsReadyWithMachineConfig`
* Avoid the `mustExist` flag. Have a `getMachineConfigPoolsByLabel(oc *exutil.CLI, mcpSelectorLabel string)`, correctly handle M`MatchExpression`there as well and do somethign like
```
func areMasterMachineConfigPoolsWithIPsec(oc *exutil.CLI) (bool, error) {
    pools , err := getMachineConfigPoolsByLabel(...)
    if err != nil {
        return false, err
    }
    return areMachineConfigPoolsReadyWithMachineConfig(pools, ...)    
}
```
and so on...

Reply:
This looks perfect and clean. Done.
Thanks @jcaamano .